ee/hogai/assistant.py

```diff
@@ -507,32 +507,34 @@ def _process_value_update(self, update: GraphValueUpdateTuple) -> list[BaseModel
 
     def _process_message_update(self, update: GraphMessageUpdateTuple) -> BaseModel | None:
         langchain_message, langgraph_state = update[1]
-        if not isinstance(langchain_message, AIMessageChunk):
-            return None
-
-        node_name: MaxNodeName = langgraph_state["langgraph_node"]
-
-        # Check for reasoning content first (for all nodes that support it)
-        if reasoning := langchain_message.additional_kwargs.get("reasoning"):
-            if reasoning_headline := self._chunk_reasoning_headline(reasoning):
-                return ReasoningMessage(content=reasoning_headline)
-
-        # Only process streaming nodes
-        if node_name not in STREAMING_NODES:
-            return None
+        # Process message chunks
+        if isinstance(langchain_message, AIMessageChunk):
+            node_name: MaxNodeName = langgraph_state["langgraph_node"]
+            # Check for reasoning content first (for all nodes that support it)
+            if reasoning := langchain_message.additional_kwargs.get("reasoning"):
+                if reasoning_headline := self._chunk_reasoning_headline(reasoning):
+                    return ReasoningMessage(content=reasoning_headline)
+
+            # Only process streaming nodes
+            if node_name not in STREAMING_NODES:
+                return None
 
-        # Merge message chunks
-        self._merge_message_chunk(langchain_message)
+            # Merge message chunks
+            self._merge_message_chunk(langchain_message)
 
-        if node_name == AssistantNodeName.MEMORY_INITIALIZER:
-            return self._process_memory_initializer_chunk(langchain_message)
+            if node_name == AssistantNodeName.MEMORY_INITIALIZER:
+                return self._process_memory_initializer_chunk(langchain_message)
 
-        # Extract and process content
-        message_content = extract_content_from_ai_message(self._chunks)
-        if not message_content:
-            return None
+            # Extract and process content
+            message_content = extract_content_from_ai_message(self._chunks)
+            if not message_content:
+                return None
 
-        return AssistantMessage(content=message_content)
+            return AssistantMessage(content=message_content)
+        # Return ready messages as is
+        elif isinstance(langchain_message, get_args(AssistantMessageUnion)):
```

Comment:
Formatting change. The only difference is this line - if it's `AIMessageChunk` - continue as before, if it's an assistant message - return the message. Got this suggestion from Em.

@Twixes need your opinion on if I should limit it to notebook messages only.

Reply:
@Twixes It has nested unions inside unions, so `get_args` is a paranoid approach :)